src/gc/gcinterface.h

```diff
@@ -357,7 +357,17 @@ class IGCHeap {
     /*
     ===========================================================================
     Allocation routines. These all call into the GC's allocator and may trigger a garbage
-    collection.
+    collection. All allocation routines return NULL when the allocation request
+    couldn't be serviced due to being out of memory.
+
+    These allocation routines should not be called with allocation requests
+    larger than:
+       32-bit                                 -> 0x7FFFFFE0
+       64-bit, GCAllowVeryLargeObjects: false -> 0x000000007FFFFFE0
```

Comment:
GCAllowVeryLargeObjects should be omitted here. It is internal limitation imposed by the CLR VM; not part of the GC interface.